dataless-classifier/src/main/java/edu/illinois/cs/cogcomp/datalessclassification/ta/DatalessAnnotatorUtils.java

```diff
@@ -0,0 +1,131 @@
+/**
+ * This software is released under the University of Illinois/Research and Academic Use License. See
+ * the LICENSE file in the root folder for details. Copyright (c) 2016
+ *
+ * Developed by: The Cognitive Computation Group University of Illinois at Urbana-Champaign
+ * http://cogcomp.cs.illinois.edu/
+ */
+package edu.illinois.cs.cogcomp.datalessclassification.ta;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+public class DatalessAnnotatorUtils {
+
+    public static Map<String, String> getLabelNameMap(String filePath) {
+        Map<String, String> labelNameMap = new HashMap<>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                if (line.isEmpty())
+                    continue;
+
+                String[] tokens = line.split("\t", 2);
+
+                String labelId = tokens[0].trim();
+                String labelName = tokens[1].trim();
+
+                labelNameMap.put(labelId, labelName);
+            }
+
+            reader.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return labelNameMap;
+    }
+
+    public static Map<String, String> getLabelDescriptionMap(String filePath) {
+        Map<String, String> labelDesriptionMap = new HashMap<>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(filePath));
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                if (line.isEmpty())
+                    continue;
+
+                String[] tokens = line.split("\t", 2);
+
+                String labelId = tokens[0].trim();
+                String labelDesc = tokens[1].trim();
+
+                labelDesriptionMap.put(labelId, labelDesc);
+            }
+
+            reader.close();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        return labelDesriptionMap;
+    }
+
+    public static Set<String> getTopNodes(String hierarchyPath) {
+        Set<String> topNodes = new HashSet<>();
+
+        try {
+            BufferedReader reader = new BufferedReader(new FileReader(hierarchyPath));
+            String line = reader.readLine();
+
+            String[] nodes = line.split("\t");
+
+            for (String node : nodes) {
+                topNodes.add(node);
+            }
+
+            reader.close();
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Throw the exception and somewhere at the top of the call stack, fail with a helpful error message.

Reply:
Done.